bindings/python/repair_wheel.py

```diff
@@ -0,0 +1,80 @@
+import glob
+import os
+import sys
+from subprocess import run
+
+HERE = os.path.abspath(os.path.dirname(__file__))
+wheel_dir, wheel_file, delocate_args = sys.argv[1:]
+wheel_dir = wheel_dir.replace(os.sep, "/")
+wheel_file = wheel_file.replace(os.sep, "/")
+
+# Ensure pyarrow.
+if "universal2" in wheel_file:
+    macos_ver = os.environ.get("MACOSX_DEPLOYMENT_TARGET", "10.3")
+    macos_ver = macos_ver.replace(".", "_")
+    target = os.path.expanduser("~/wheels")
+    run(
+        [
+            sys.executable,
+            "-m",
+            "pip",
+            "install",
+            "--platform",
+            f"macosx_{macos_ver}_universal2",
+            "--upgrade",
+            "--target",
+            target,
+            "--only-binary=:all:",
+            "pyarrow",
+        ]
+    )
+    # Allow the installed pyarrow library to be imported.
+    sys.path.insert(0, target)
+else:
+    run([sys.executable, "-m", "pip", "install", "pyarrow"])
+import pyarrow as pa  # noqa
+
+libbson = os.environ.get("LIBBSON_INSTALL_DIR", os.path.join(HERE, "libbson"))
+libbson = os.path.abspath(libbson)
+if os.name == "nt":
+    libbson_lib = glob.glob(os.path.join(libbson, "bin"))
+else:
+    libbson_lib = glob.glob(os.path.join(libbson, "lib*"))
+extra_paths = pa.get_library_dirs() + libbson_lib
+extra_path = os.path.pathsep.join([a.replace(os.sep, "/") for a in extra_paths])
+
+
+def append_os_variable(name):
+    if os.environ.get(name):
+        os.environ[name] = os.environ[name] + os.pathsep + extra_path
```

Comment:
Can you make extra_path an argument? Mixing a global in this method makes it harder to read imo.

Reply:
Done